src/lumigo_tracer/lumigo_utils.py

```diff
@@ -337,6 +347,15 @@ def report_json(region: Optional[str], msgs: List[dict], should_retry: bool = Tr
     return duration
 
 
+def write_spans_to_file(to_send):
+    file_name = f"{str(hashlib.md5(to_send).hexdigest())}_single"
+    Path(EXTENSION_DIR).mkdir(parents=True, exist_ok=True)
+    file_path = os.path.join(EXTENSION_DIR, file_name)
+    get_logger().info(f"writing spans to file {file_path}")
+    with open(file_path, "wb") as spans_file:
+        spans_file.write(to_send)
```

Comment:
I think that this part should be inside `lumigo_safe_execute` because accessing the files-system can result in an error.